app/src/main/java/org/mozilla/fenix/components/metrics/GleanMetricsService.kt

```diff
@@ -430,12 +430,17 @@ class GleanMetricsService(private val context: Context) : MetricsService {
         if (initialized) return
         initialized = true
 
+        // We have to initialize Glean *on* the main thread, because it registers lifecycle
+        // observers. However, the activation ping must be sent *off* of the main thread,
+        // because it calls Google ad APIs that must be called *off* of the main thread.
+        // These two things actually happen in parallel, but that should be ok because Glean
```

Comment:
Yes, I think this should be covered, especially because we call `setUploadEnabled` before. I think we should manually check that the activation ping gets sent, if possible.